src/context/UserContext.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  createContext,
+  useContext,
+  useReducer,
+  ReactNode,
+  Dispatch,
+} from 'react';
+
+interface UserState {
+  name: string;
+  email: string;
+  isLoggedIn: boolean;
+  currency: string;
```

Comment:
There's a limited list of currencies, and each of them has standard short code. In cases like this, using a string would eventually result in misspelling (and other) errors. 
To avoid it, it's better to create an `enum` that will hold Currency values that we support in the application. You can see the list here: https://www.figma.com/design/6KSlLFezGkD8uZXhuTsx3j/Montra---Expense-Tracker-UI-Kit-(Community)?node-id=768-3696&node-type=FRAME&t=qKzsJLWaWTjDg99b-0

Reply:
Fixed